src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineLinuxLauncher.java

```diff
@@ -127,12 +127,14 @@ protected void launch(ComputeEngineComputer computer, TaskListener listener, Ins
       scp.put(Jenkins.get().getJnlpJars("agent.jar").readFully(), "agent.jar", jenkinsDir);
 
       // Confirm Java is installed
-      if (!testCommand(computer, conn, "java -fullversion", logger, listener)) {
-        logWarning(computer, listener, "Java is not installed.");
+      String javaExecPath = node.getJavaExecPathOrDefault();
+      if (!testCommand(
+          computer, conn, String.format("%s -fullversion", javaExecPath), logger, listener)) {
+        logWarning(computer, listener, String.format("%s is not installed.", javaExecPath));
       }
 
       // TODO: allow jvmopt configuration
-      String launchString = "java -jar " + jenkinsDir + "/agent.jar";
+      String launchString = String.format("%s -jar ", javaExecPath) + jenkinsDir + "/agent.jar";
```

Comment:
There's a lot of duplication here between these launch methods, I think it might be good to pull out one or two helper methods into ComputeEngineComputerLauncher for checking the java install and generating a launch string. 
To tackle the issue of pathing differences, I would recommend having a base class method to retrieve the agent path that gets overridden in these subclasses, one returning `/agent.jar` and the other `\\agent.jar`, or something along those lines.

Reply:
Understood, the TODOs you added are enough for this PR.